dotnet/simple-app/Program.cs

```diff
@@ -0,0 +1,284 @@
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System.Text;
+using ImmuDB;
+using ImmuDB.Exceptions;
+using ImmuDB.SQL;
+
+namespace simple_app;
+
+class Program
+{
+    private static string immudbServerAddress = "localhost";
+    
+    public static async Task Main(string[] args)
+    {
+        string? envAddr = Environment.GetEnvironmentVariable("IMMUDB_ADDRESS");
+        if(!string.IsNullOrEmpty(envAddr)) {
+            immudbServerAddress = envAddr;
+        }
+        await OpenConnectionExample();
+        await AnotherOpenConnectionExample();
+        await GetSetScanUsageExample();
+        await SetAllGetAllExample();
+        await SqlUsageExample();
+
+        SyncOpenConnectionExample();
+        SyncSqlUsageExample();
+
+        await ImmuClient.ReleaseSdkResources();
+    }
+
+    private static async Task OpenConnectionExample()
+    {
+        var client = ImmuClient.NewBuilder().WithServerUrl(immudbServerAddress).Build();
+        await client.Open("immudb", "immudb", "defaultdb");
+
+        string key = "hello";
+
+        try
+        {
+            await client.VerifiedSet(key, "immutable world!");
+
+            // Getting it back, by key (in a verified way that reports any tampering if it happened).
+            Entry entry = await client.VerifiedGet(key);
+            Console.WriteLine($"{key}, {entry.ToString()}");
+        }
+        catch (VerificationException e)
+        {
+            // VerificationException means Data Tampering detected!
+            // This means the history of changes has been tampered.
+            Console.WriteLine(e.ToString());
+        }
+        await client.Close();
+    }
+    
+    private static void SyncOpenConnectionExample()
+    {
+        var client = ImmuClientSync.NewBuilder().WithServerUrl(immudbServerAddress).Build();
+        client.Open("immudb", "immudb", "defaultdb");
+
+        string key = "hello";
+
+        try
+        {
+            client.VerifiedSet(key, "immutable world!");
+
+            // Getting it back, by key (in a verified way that reports any tampering if it happened).
+            Entry entry = client.VerifiedGet(key);
+            Console.WriteLine($"{key}, {entry.ToString()}");
+        }
+        catch (VerificationException e)
+        {
+            // VerificationException means Data Tampering detected!
+            // This means the history of changes has been tampered.
+            Console.WriteLine(e.ToString());
+        }
+        client.Close();
+    }
+
+    private static async Task AnotherOpenConnectionExample()
+    {
+        var client = await ImmuClient.NewBuilder().WithServerUrl(immudbServerAddress).Open();

```

Comment:
How does it work? Where the username and password come from ?

Reply:
There are defaults set by the builder. The aim was to make it as simple as possible.